### PR TITLE
add system tags

### PR DESCRIPTION
**Pull request by eherget**
This adds systems tags support and some additional unit tests.

### CODE REVIEW

File: kerlescan/profile_parser.py [hunk 1]
```diff
@@ -94,8 +94,25 @@ def _parse_interface(name):
...
+    def _parse_tags(tags):
+        """
+        helper method to convert tag lists to facts. We output a
+        fact for each unique tag (combination of namespace and key).
+        The value is a comma-separated ordered list of all values
+        for the tag.
+        """
+        tag_dict = {}
+        for tag in tags:
+            tag_name = "{}.{}".format(tag["namespace"], tag["key"])
```
Comment (beav) [inline kerlescan/profile_parser.py:106]: that is cool, I didnt know of this method before

File: kerlescan/inventory_service_interface.py [hunk 2]
```diff
@@ -22,7 +23,9 @@ def ensure_correct_system_count(system_ids_requested, result):
         )
 
 
-def interleave_systems_and_profiles(systems_result, system_profiles_result):
+def interleave_systems_and_profiles(
+    systems_result, system_profiles_result, system_tags_result
+):
```
Comment (beav) [inline kerlescan/inventory_service_interface.py:28]: would it be useful if inventory had a "give me everything" API? so you don't have to call it 3x

Comment (eherget) [reply kerlescan/inventory_service_interface.py:28]: Yes, it definitely would.  Maybe tags should/could be in the metadata, although we will want them in the system profile.  I still have to look at what needs to be done for HSP.  Do we write the historical profiles?  I also need to look at system-baseline-backend as it calls parse_profile in one place (views/v1.py)

Comment (eherget) [reply kerlescan/inventory_service_interface.py:28]: BTW - I saw a discussion yesterday surrounding the all_installed_packages enhancement request that maybe the inventory API should provide a way for the caller to request what data they want in the response.

File: kerlescan/profile_parser.py (hunk 1 above)
Comment (eherget) [reply kerlescan/profile_parser.py:106]: You mean the "{}".format() stuff?  It's pretty neat.  I think you can also do named substitution and pass in a dict with the values like this:
myvals = {"name": "eric", "host": "localhost"}
retval = "{name} some text {host}".format(myvals)
I'd check the docs first, however ;-)

Comment (eherget) [reply kerlescan/profile_parser.py:106]: Just checked... not a dict, but a list of key=value.  And can do some formatting, too.  {price:.2f} for example.

File: kerlescan/inventory_service_interface.py [hunk 3]
```diff
@@ -67,6 +70,13 @@ def interleave_systems_and_profiles(systems_result, system_profiles_result):
         system_with_profile["system_profile"]["fqdn"] = system["fqdn"]
         system_with_profile["system_profile"]["updated"] = system["updated"]
 
+        # now add tags if any exist for the system.
+        # system_tags_result is a dict keyed on uuid.
+        if system["id"] in system_tags_result and system_tags_result[system["id"]]:
```
Comment (beav) [inline kerlescan/inventory_service_interface.py]: Is this equivalent?
```python
if system_tags_result.get(system["id"], None)
```

**Comment by beav** [comment]
this looks good to me. Thanks!

File: kerlescan/inventory_service_interface.py (hunk 3 above)
Comment (eherget) [reply kerlescan/inventory_service_interface.py]: Yes, I think it is!  I'll make the change.

Comment (eherget) [reply kerlescan/inventory_service_interface.py]: ...and the default return value is None, so no need for ", None"

**Review by beav — APPROVED** [review]

**Review by CodeHeeler — APPROVED** [review]
LGTM, and thanks for adding the tests as well.

**Pull request merged**